stix_shifter_modules/guardium/stix_transmission/ping_connector.py

```diff
@@ -1,25 +1,25 @@
 from stix_shifter_utils.modules.base.stix_transmission.base_ping_connector import BasePingConnector
-import json
 from stix_shifter_utils.utils.error_response import ErrorResponder
-
+from stix_shifter_utils.utils import logger
 
 class PingConnector(BasePingConnector):
     def __init__(self, api_client):
         self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
 
     def ping_connection(self):
-        # Implemented for Guardium using fetch_token
-        response = self.api_client.ping_box()
-        response_code = response.code
-
-        response_dict = json.loads(response.read())
-
-        return_obj = dict()
-        return_obj['success'] = False
+        try:
+            response_dict = self.api_client.ping_data_source()
+            # response_dict = {'code': 1010, 'message': 'remote system error message'} # <-- simulate error in response to test error mapping
+            response_code = response_dict["code"]
 
-        if len(response_dict) > 0 and response_code == 200:
-            return_obj['success'] = True
-            #return_obj["search_id"] = response_dict.get('search_id',"NA")
-        else:
-            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
-        return return_obj
+            # Construct a response object
+            return_obj = dict()
+            if response_code == 200            :
+                return_obj['success'] = True
+            else:
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+            return return_obj
+        except Exception as err:
+            self.logger.error('error when pinging datasource {}:'.format(err))
+            raise
```

Comment:
nothing major. just for better maintainability. I would suggest to move this file `stix_shifter_modules/guardium/stix_transmission/guard_utils.py` inside `stix_shifter_modules/guardium/stix_transmission/utils` folder

Reply:
if it is not major, Id like to leave it there.